src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -381,12 +381,18 @@ public void setName(String name) {
      * @return interned string
      */
     protected String getCachedString(String str) {
-        WeakReference<String> internedString = stringCache.get(str);
-        if ( internedString == null || internedString.get() == null) {
-            internedString = new WeakReference<>(str);
-            stringCache.put(str, internedString);
+        if (!vcfIndexUseStringsCache)
+            return str;
+        WeakReference<String> stringWeakReference = stringCache.get(str);
```

Comment:
As I said before, I dislike the use of WeakReferences for this for two reasons:
1. It leaves half the problem there (the keys). We had a memory "explosion" with a map that has keys and values. Removing the values will only make 1/2 the problem go away
2. A WeakReference gives us very little control over when the objects are GCd. I'd prefer to see a LFU (or even preferably LFUDA) cache be used instead.